### DISCUSSION TITLE
Use middleware from server route

### DISCUSSION THREAD

**Discussion by GoncaloPataca** [nuxt/nuxt#17323] [General]
Is there a way to invoke a middleware from a specific api route?

Let's say that I have a file checkToken.js which is used to verify the user's jwt. This middleware can either return an access forbidden response, or return nothing, letting the request proceed to the next handler.
Then let's say I have an actual route server/api/data.js, from where I want to make a call to the checkToken.js middleware. If the jwt is invalid the response is returned from checkToken.js, else it is returned from data.js.

I do not want to put checkToken.js in the server/middleware directory, as it will be executed from both client and server side, and makes it difficult to specify which routes use which middleware. Is there a way to achieve this?

**Comment by alpeshvmodi** [comment]
I am also searching for the solution. Is there any help available for this? 
@danielroe Do you have any clue for this?
